client/src/main/scala/com/all4journey/webapp/pages/PlacesFormJsImpl.scala

```diff
@@ -44,58 +46,67 @@ object PlacesFormJsImpl extends PlacesFormJs with AddressTypePickler {
-      val option = dom.document.createElement("option")
-      option.textContent = stateItem.description
-      option.setAttribute("value", stateItem.id)
-      stateDropdown.appendChild(option)
+      val option = buildDropwDownOption(stateItem.description, stateItem.id)
+      $(".partOfStateList").append(option)
     }
+
   }
 
   @JSExport
-  def buildPlacesDropDown(addresses: Seq[Address]): Unit = {
+  def buildPlacesForm(addressToDisplay: Option[Address], addresses: Seq[Address]): Unit = {
 
-    placeNames.clear
+    val homeAddressOption = addresses.find(address => address.addressType == HomeAddressType)
+    homeAddressOption.foreach(
+      homeAddress => AddressForm.refreshFields(homeAddress)
+    )
 
-    val placesDropdown = dom.document.getElementById("places")
+    val places = addresses.filter(
+      address => address.addressType == PlaceAddressType
+    )
 
-    // find the HOME place and add it to the top
-    var listWithoutHomeType = new ListBuffer[Address]()
-      for (addressItem <- addresses) {
-        if (addressItem.addressType.equals(HomeAddressType)) {
-          buildDropwDownOption(placesDropdown, addressItem.placeName, addressItem.id)
-
-          AddressForm.refreshFields(addressItem)
-        }
+    placeNames.clear
+    places.map(
+      placeNames += _.placeName
+    )
 
-        else {
-          listWithoutHomeType += addressItem
-        }
+    val placesDropdown = dom.document.getElementById("places")
 
-        placeNames += addressItem.placeName
-      }
+    places.foreach(addressItem =>
+      buildDropwDownOption(placesDropdown, addressItem.placeName, addressItem.id)
+    )
 
+    buildDropwDownOption(placesDropdown, AddNewPlaceIndicator, "0")
 
-    for (addressItem <- listWithoutHomeType) {
-      buildDropwDownOption(placesDropdown, addressItem.placeName, addressItem.id)
+    val defaultPlaceOption = addressToDisplay match {
+      case Some(address) if address.addressType == PlaceAddressType => addressToDisplay
+      case _ => places.headOption
     }
 
-    buildDropwDownOption(placesDropdown, AddNewPlaceIndicator, "0")
+    defaultPlaceOption.foreach(defaultPlace => {
+      AddressForm.refreshFields(defaultPlace)
+      $("#placeName").value(defaultPlace.placeName)
+      $("#places").value(defaultPlace.id).change()
+    })
 
   }
 
   @JSExport
   def buildDropwDownOption(dropDown: Element, textContent: String, optionIndex: String): Unit = {
+    val option = buildDropwDownOption(textContent, optionIndex)
+    dropDown.appendChild(option)
+  }
+
+  @JSExport
+  def buildDropwDownOption(textContent: String, optionIndex: String): Element = {
     val option = dom.document.createElement("option")
     option.textContent = textContent
     option.setAttribute("value", optionIndex)
-    dropDown.appendChild(option)
+    option
   }
 
   @JSExport
-  def placesForm(defaultAddress: Option[Address]) = div(cls := "container")(
+  def placesForm: TypedTag[Div] = div(cls := "container")(
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Method is longer than 50 lines](https://www.codacy.com/app/rjkernick/all4journey/file/2737533538/issues/source?bid=3246460&fileBranchId=3352852#l109)